docs/common_hal.md

```diff
@@ -0,0 +1,70 @@
+# Adding `nativeio` support to other ports
```

Comment:
I did use the ones that made sense. :-) I appreciate all of the suggestions. I want to make the docs as friendly as possible.